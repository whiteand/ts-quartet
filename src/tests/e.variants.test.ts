import { e as v } from "..";
import { testValidatorImpure } from "./testValidatorImpure";

describe("v([...])", () => {
  test("v([])", () => {
    testValidatorImpure(
      v([]),
      [],
      [null, false, [], {}, 1, 0, NaN, undefined, true]
    );
  });
  test("v([1])", () => {
    testValidatorImpure(
      v([1]),
      [1],
      [null, false, [], {}, 0, NaN, undefined, true]
    );
  });
  test("v([1, 2])", () => {
    testValidatorImpure(
      v([1, 2]),
      [1, 2],
      ["2", "1", null, false, [], {}, 0, NaN, undefined, true]
    );
  });
  test("v([1, '2'])", () => {
    testValidatorImpure(
      v([1, "2"]),
      [1, "2"],
      [2, "1", null, false, [], {}, 0, NaN, undefined, true]
    );
  });
  test("v([true, false])", () => {
    let flag = true;
    const validator = v([
      v.any,
      v.custom(() => {
        flag = false;
        return false;
      })
    ]);
    validator(1);
    expect(flag).toBeTruthy();
  });
});
