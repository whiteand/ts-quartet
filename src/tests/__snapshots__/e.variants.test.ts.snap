// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v([...]) v([]) 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": null,
  },
]
`;

exports[`v([...]) v([]) 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": false,
  },
]
`;

exports[`v([...]) v([]) 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": Array [],
  },
]
`;

exports[`v([...]) v([]) 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": Object {},
  },
]
`;

exports[`v([...]) v([]) 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": 1,
  },
]
`;

exports[`v([...]) v([]) 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": 0,
  },
]
`;

exports[`v([...]) v([]) 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": NaN,
  },
]
`;

exports[`v([...]) v([]) 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": undefined,
  },
]
`;

exports[`v([...]) v([]) 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Never",
    },
    "value": true,
  },
]
`;

exports[`v([...]) v([1, '2']) 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": 2,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": 2,
  },
]
`;

exports[`v([...]) v([1, '2']) 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": "1",
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": "1",
  },
]
`;

exports[`v([...]) v([1, '2']) 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": null,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": null,
  },
]
`;

exports[`v([...]) v([1, '2']) 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": false,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": false,
  },
]
`;

exports[`v([...]) v([1, '2']) 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Array [],
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": Array [],
  },
]
`;

exports[`v([...]) v([1, '2']) 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Object {},
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": Object {},
  },
]
`;

exports[`v([...]) v([1, '2']) 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": 0,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": 0,
  },
]
`;

exports[`v([...]) v([1, '2']) 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": NaN,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": NaN,
  },
]
`;

exports[`v([...]) v([1, '2']) 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": undefined,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": undefined,
  },
]
`;

exports[`v([...]) v([1, '2']) 10`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": true,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": "2",
    },
    "value": true,
  },
]
`;

exports[`v([...]) v([1, 2]) 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": "2",
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": "2",
  },
]
`;

exports[`v([...]) v([1, 2]) 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": "1",
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": "1",
  },
]
`;

exports[`v([...]) v([1, 2]) 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": null,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": null,
  },
]
`;

exports[`v([...]) v([1, 2]) 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": false,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": false,
  },
]
`;

exports[`v([...]) v([1, 2]) 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Array [],
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": Array [],
  },
]
`;

exports[`v([...]) v([1, 2]) 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Object {},
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": Object {},
  },
]
`;

exports[`v([...]) v([1, 2]) 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": 0,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": 0,
  },
]
`;

exports[`v([...]) v([1, 2]) 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": NaN,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": NaN,
  },
]
`;

exports[`v([...]) v([1, 2]) 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": undefined,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": undefined,
  },
]
`;

exports[`v([...]) v([1, 2]) 10`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": true,
  },
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 2,
    },
    "value": true,
  },
]
`;

exports[`v([...]) v([1]) 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": null,
  },
]
`;

exports[`v([...]) v([1]) 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": false,
  },
]
`;

exports[`v([...]) v([1]) 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Array [],
  },
]
`;

exports[`v([...]) v([1]) 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": Object {},
  },
]
`;

exports[`v([...]) v([1]) 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": 0,
  },
]
`;

exports[`v([...]) v([1]) 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": NaN,
  },
]
`;

exports[`v([...]) v([1]) 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": undefined,
  },
]
`;

exports[`v([...]) v([1]) 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Primitive",
      "value": 1,
    },
    "value": true,
  },
]
`;
